src/main/java/com/networknt/schema/OneOfValidator.java

```diff
@@ -174,26 +176,40 @@ public OneOfValidator(String schemaPath, JsonNode schemaNode, JsonSchema parentS
 
                 numberOfValidSchema++;
             }
+            childErrors.addAll(schemaErrors);
         }
 
 
         // ensure there is always an "OneOf" error reported if number of valid schemas is not equal to 1.
-        if (numberOfValidSchema != 1) {
-            errors = Collections.singleton(buildValidationMessage(at, ""));
+        if(numberOfValidSchema > 1){
+            final ValidationMessage message = getBothSchemasValidErrorMsg(at);
+            if(failFast)
+                throw new JsonSchemaException(message);
+            errors.add(message);
+        }
+        // ensure there is always an "OneOf" error reported if number of valid schemas is not equal to 1.
+        else if (numberOfValidSchema < 1) {
+            if (!childErrors.isEmpty()) {
+                errors.addAll(childErrors);
+            }
+            if(failFast){
+                throw new JsonSchemaException(errors.toString());
+            }

```

Comment:
Missing some spaces between "if" and parathesis

Reply:
Thank you @guillomep. I have added spaces as you suggested.